engine/src/flutter/shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -544,6 +544,50 @@ public void setSystemUiModeListener_overlaysAreVisible() {
     verify(mockPlatformChannel).systemChromeChanged(true);
   }
 
+  @SuppressWarnings("deprecation")
+  @Config(sdk = API_LEVELS.API_30)
+  @Test
+  public void doEnableEdgeToEdgeWithoutViewOnNewerSdk() {
+    View fakeDecorView = mock(View.class);
+    Window fakeWindow = mock(Window.class);
+    Activity mockActivity = mock(Activity.class);
+    when(fakeWindow.getDecorView()).thenReturn(fakeDecorView);
+    when(mockActivity.getWindow()).thenReturn(fakeWindow);
+    PlatformPlugin platformPlugin = new PlatformPlugin(mockActivity, mockPlatformChannel);
+
+    platformPlugin.mPlatformMessageHandler.showSystemUiMode(
+        PlatformChannel.SystemUiMode.EDGE_TO_EDGE);
+    // Does not use view.
+    verify(fakeDecorView, never())
+        .setSystemUiVisibility(
+            View.SYSTEM_UI_FLAG_LAYOUT_STABLE
```

Comment:
Nit: We could check that `setSystemUiVisibility` doesn't get called with any flags (not just these specific ones) just in case.